### PR TITLE
[GH-1059] Fix empty placeholders in the card dialog

### LINKED ISSUE #1059: Bug: Number property should show "Empty" placeholder in card editor
**Steps to reproduce the behavior:**
1. Open a card with a number property that is not set
2. The field placeholder is blank

**Expected behavior:**
* The field placeholder should show the text "Empty" in the card editor for all property types
* However, the placeholder should be blank in tables

**Screenshots:**
Number empty

**Platform:**
 - Main branch build, Aug 20, 2021

**Additional context:**
This is a regression caused by #966, see [thread](https://community.mattermost.com/core/pl/aaz64xgkspri3ff8k7cxoatbmo).

The property value display logic needs to be refactored, probably with a "showEmptyPlaceholder" flag. True for the card editor, and false for tables.

### PR DESCRIPTION
**Pull request by kamre**
#### Summary
Show empty placeholder for values in the card dialog. At the same time empty placeholders are not used for table cell, kanban or gallery cards.

Here is how it looks in the card dialog:
image

In kanban card:
image

In the table view:
TableViewRow

#### Ticket Link
Fixes #1059

### CODE REVIEW

**Comment by harshilsharma63** [comment]
@kamre awesome work as always 👍 

One change though. The empty placeholder looks nice for text and other text-based properties (email, URL, etc) but the styling drifts away for some special property types. See the screenshot attached below-
 
Screenshot 2021-09-06 at 5 44 15 PM

It will look really nice and consistent if the empty palceholder looks consistent across property types. DO you think this is doable?

**Comment by kamre** [comment]
@harshilsharma63 ok, I will look into this. But there is also more generic issue #1035 about styles.

**Comment by kamre** [comment]
@harshilsharma63 updated styles for empty placeholders:
image

File: webapp/src/widgets/valueSelector.tsx [hunk 1]
```diff
@@ -114,6 +114,10 @@ const valueSelectorStyle = {
         padding: '0 8px',
         overflow: 'unset',
     }),
+    placeholder: (provided: CSSObject): CSSObject => ({
+        ...provided,
+        color: 'rgba(var(--center-channel-color-rgb), 0.4)',
+    }),
```
Comment (kamre) [inline webapp/src/widgets/valueSelector.tsx:120]: May be this style for placeholder should be moved to [getSelectBaseStyle](https://github.com/mattermost/focalboard/blob/main/webapp/src/theme.ts#L242)

**Review by harshilsharma63 — APPROVED** [review]
LGTM!

File: webapp/src/widgets/valueSelector.tsx (hunk 1 above)
Comment (harshilsharma63) [reply webapp/src/widgets/valueSelector.tsx:120]: If we move it to `getSelectBaseStyle` we have to specify select styling for places other than card properties. Either way, we'll end up having something specified in base and it being overridden in uses, so I think what we have here is fine.

**Comment by kamre** [comment]
@chenilim do we need any additional changes for this pull request?

**Comment by chenilim** [comment]
Apologies for the delay @kamre! I've been behind on my GitHub alerts. I've smoke tested it and the fix looks good to me. Unfortunately, it looks like you need a manual merge conflict resolve.

For future reference, we just need one developer approval to merge - so whoever approves it, just go ahead and merge if there are no further changes needed.

**Review by chenilim — APPROVED** [review]
LGTM. Thanks!

**Comment by chenilim** [comment]
Was able to manually merge. Thanks!

**Pull request merged**